src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -155,6 +155,40 @@ public IActionResult DrainStatus([FromServices] IScriptHostManager scriptHostMan
             }
         }
 
+        [HttpPost]
+        [Route("admin/host/resume")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public async Task<IActionResult> Resume([FromServices] IScriptHostManager scriptHostManager)
+        {
+            _logger.LogDebug($"Received request to resume a draining host - host status: {scriptHostManager?.State.ToString()}");
+
+            if (scriptHostManager?.State != ScriptHostState.Running)
+            {
+                _logger.LogWarning("The host is not in a state where we can resume. The host's state needs to be 'Running' and drain mode must be enabled for resume to work.");
```

Comment:
I think you should combine the check here with the one below - basically try to access IDrainMode manager and check drain status / state all in one go and return the error if needed in a single check?

Reply:
Great point about the error message being out of sync, I'll move thing around!

Reply:
Updates look good to me